standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -7791,7 +7791,7 @@ public List<String> get_role_names() throws TException {
       ret = getMS().listRoleNames();
       return ret;
     } catch (Exception e) {
-      throw handleException(e).throwIfInstance(MetaException.class).defaultRuntimeException();
```

Comment:
I'm afraid that convert other exceptions to TException would not retry in case of JDOException, etc.

Reply:
Thanks for your concern @dengzhhu653.
Converting to `TException` should be OK because it is actually converted to a `MetaException`:
```java
// standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java
  public TException defaultTException() {
    if (e instanceof TException) {
      return (TException) e;
    }
    return newMetaException(e);
  }
```

Reply:
Got your point. we consider that some JDOExceptions can be resolved by just retrying, such as this rare [exception](https://github.com/brettwooldridge/HikariCP/issues/1288).